File: CS/WebSite/App_Code/Invoice.cs
using System;
using System.Collections.Generic;

[Serializable]
public class Invoice : IEquatable<Invoice> {
    public int Id { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }

    public Invoice(int id, string description, decimal price) {
        Id = id;
        Description = description;
        Price = price;
    }

    public Invoice() { }

    public bool Equals(Invoice other) {
        return this.Id.Equals(other.Id);
    }
}

File: VB/WebSite/App_Code/Invoice.vb
Imports System
Imports System.Collections.Generic

<Serializable> _
Public Class Invoice
    Implements IEquatable(Of Invoice)

    Public Property Id() As Integer
    Public Property Description() As String
    Public Property Price() As Decimal

    Public Sub New(ByVal id As Integer, ByVal description As String, ByVal price As Decimal)
        Me.Id = id
        Me.Description = description
        Me.Price = price
    End Sub

    Public Sub New()
    End Sub

    Public Function Equals(ByVal other As Invoice) As Boolean Implements IEquatable(Of Invoice).Equals
        Return Me.Id.Equals(other.Id)
    End Function
End Class